<a href="https://colab.research.google.com/github/pabloecheg/Business-Intelligence/blob/main/Reto_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Juan Pablo Villamizar, Pablo Echeverri

In [ ]:
import sqlite3

In [ ]:
import pandas as pd
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
nvidia = pd.read_excel('/content/drive/MyDrive/INTRO BI/RETO 2/NVIDIA_consolidado.xlsx', sheet_name= None, header=0)
apple = pd.read_excel('/content/drive/MyDrive/INTRO BI/RETO 2/Apple_consolidado.xlsx', sheet_name= None, header=0)

Miramos las claves

In [ ]:
nvidia.keys()

dict_keys(['BALANCE GENERAL', 'RATIOS DE RENTABILIDAD', 'ESTADO DE RESULTADOS', 'ESTADO DE FLUJO DE EFECTIVO'])

In [ ]:
apple.keys()

dict_keys(['BALANCE  GENERAL', 'RATIOS DE RENTABILIDAD', 'ESTADO DE RESULTADOS', 'ESTADO DE FLUJO DE EFECTIVO'])

# NVIDIA

Creamos la base de datos

In [ ]:
con = sqlite3.connect('nvidia.db')
cursor = con.cursor()

Pasamos la informacion del diccionario a la base de datos

In [ ]:
for hoja, nvidia in nvidia.items():
  nvidia.to_sql(hoja, con, if_exists='replace', index=False)

In [ ]:
#Confirmamos los datos
con.commit()

In [ ]:
#Visualizamos los nombres de las tablas en formato consulta
consulta = ''' SELECT name
               FROM sqlite_master
               WHERE type = 'table' '''
#Ejecutamos la consulta
cursor.execute(consulta)
#Printeamos el resultado de la consulta
print(cursor.fetchall())

[('BALANCE GENERAL',), ('RATIOS DE RENTABILIDAD',), ('ESTADO DE RESULTADOS',), ('ESTADO DE FLUJO DE EFECTIVO',)]


Miramos la informacion de la tabla Ratios de Rentabilidad

In [ ]:
consulta = ''' SELECT *
              FROM "RATIOS DE RENTABILIDAD" '''
#Ejecutamos la consulta
cursor.execute(consulta)
#Traemos el resultado de la consulta
Ratios = cursor.fetchall()
for fila in Ratios:
  print(fila)

(2020, 0.1615, 0.1615, 0.2291, 0.2291, 0.1831, 0.2561, 0.6548, 0.2605, 0.2956, 0.1643)
(2021, 0.1505, 0.1505, 0.2564, 0.2564, 0.1824, 0.2598, 0.6893, 0.272, 0.3376, 0.1575)
(2022, 0.2207, 0.2207, 0.3665, 0.3665, 0.2546, 0.3623, 0.6929, 0.3771, 0.4167, 0.2297)
(2023, 0.1061, 0.1061, 0.1976, 0.1976, 0.1206, 0.1619, 0.6265, 0.1548, 0.2138, 0.1014)


Miramos la informacion de la tabla Estado de Flujo de Efectivo

In [ ]:
consulta = ''' SELECT *
              FROM "ESTADO DE FLUJO DE EFECTIVO" '''
#Ejecutamos la consulta
cursor.execute(consulta)
#Traemos el resultado de la consulta
Efectivo = cursor.fetchall()
for fila in Efectivo:
  print(fila)

(2020, 4761000, 2796000, 1116000, 406000, -25000, 597000, -233000, 194000, 100000, 77000, 849000, 6145000, -792000, 10114000, 782000, 10896000, -489000)
(2021, 5822000, 4332000, 113000, 1710000, -612000, -524000, -550000, 363000, 120000, -394000, 1377000, -19675000, 3804000, -10049000, 10896000, 847000, -1128000)
(2022, 9108000, 9752000, -2595000, 1737000, -563000, -774000, -2215000, 568000, 367000, -1715000, 1951000, -9830000, 1865000, 1143000, 847000, 1990000, -976000)
(2023, 5641000, 4368000, -2827000, 2243000, -699000, -2554000, 822000, -551000, -571000, -1517000, 4100000, 7375000, -11617000, 1399000, 1990000, 3389000, -1833000)


Pasamos el resultado de las consultas a pandas

In [ ]:
columnas = [descripcion[0] for descripcion in cursor.description]
df_carta = pd.DataFrame(Ratios, columns=columnas)
df_carta

Año  Rendimiento Sobre Los Activos (ROA)  \
0  2020                               0.1615   
1  2021                               0.1505   
2  2022                               0.2207   
3  2023                               0.1061   

   Rendimiento Sobre Activos Anualizado  \
0                                0.1615   
1                                0.1505   
2                                0.2207   
3                                0.1061   

   Rendimiento Sobre El Patrimonio (ROE)  \
0                                 0.2291   
1                                 0.2564   
2                                 0.3665   
3                                 0.1976   

   Rendimiento Sobre El Patrimonio Anualizado  \
0                                      0.2291   
1                                      0.2564   
2                                      0.3665   
3                                      0.1976   

   Rendimiento Sobre El Capital Empleado  Margen Neto  \
0                                 0.1831       0.2561   
1                                 0.1824       0.2598   
2                                 0.2546       0.3623   
3                                 0.1206       0.1619   

   Margen De Ganancia Bruta  Margen Operacional  Margen Ebitda  ROA Operativo  
0                    0.6548              0.2605         0.2956         0.1643  
1                    0.6893              0.2720         0.3376         0.1575  
2                    0.6929              0.3771         0.4167         0.2297  
3                    0.6265              0.1548         0.2138         0.1014

In [ ]:
columnas = [descripcion[0] for descripcion in cursor.description]
df_carta = pd.DataFrame(Efectivo, columns=columnas)
df_carta

Año  \
0  2020   
1  2021   
2  2022   
3  2023   

   Flujo neto de efectivo por (utilizados en) actividades de explotación  \
0                                            4761000                       
1                                            5822000                       
2                                            9108000                       
3                                            5641000                       

   Utilidad Neta  Efectivo generado por las operaciones  \
0        2796000                                1116000   
1        4332000                                 113000   
2        9752000                               -2595000   
3        4368000                               -2827000   

   Ajustes por: Depreciación y deterioro de propiedad, planta y equipo  \
0                                             406000                     
1                                            1710000                     
2                                            1737000                     
3                                            2243000                     

   Ajustes por: Amortización de activos intangibles  Cambios en: Inventarios  \
0                                            -25000                   597000   
1                                           -612000                  -524000   
2                                           -563000                  -774000   
3                                           -699000                 -2554000   

   Cambios en: comerciales y otras cuentas a cobrar  \
0                                           -233000   
1                                           -550000   
2                                          -2215000   
3                                            822000   

   Cambios en: comerciales y otras cuentas a pagar  \
0                                           194000   
1                                           363000   
2                                           568000   
3                                          -551000   

   Cambios en: ingresos diferidos  Cambios en: Otras variaciones  \
0                          100000                          77000   
1                          120000                        -394000   
2                          367000                       -1715000   
3                         -571000                       -1517000   

   Otro flujo de efectivo de actividades operativas  \
0                                            849000   
1                                           1377000   
2                                           1951000   
3                                           4100000   

   Flujo neto de efectivo de (utilizadas en) actividades de inversión  \
0                                            6145000                    
1                                          -19675000                    
2                                           -9830000                    
3                                            7375000                    

   Flujo neto de efectivo de (utilizados en) actividades de financiación  \
0                                            -792000                       
1                                            3804000                       
2                                            1865000                       
3                                          -11617000                       

   Aumento (disminución) neto en efectivo y equivalentes de efectivo  \
0                                           10114000                   
1                                          -10049000                   
2                                            1143000                   
3                                            1399000                   

   Efectivo al inicio del período  Efectivo al final del período    CAPEX  
0                          782000                       10896000  -489000  
1                        10896000                         847000 -1128000  

## Consultas

Cual es la utilidad neta total de NVIDIA del 2020 al 2023?

In [ ]:
consulta = ''' SELECT SUM("Utilidad Neta") AS "Utilidad Neta Total"
               FROM "ESTADO DE FLUJO DE EFECTIVO"
               '''

#Ejecutamos la consulta
cursor.execute(consulta)
#Traemos el resultado de la consulta
Utilidad = cursor.fetchall()
for fila in Utilidad:
  print(fila)

(21248000,)


Que año tiene mas rendimiento sobre los activos (ROA)

In [ ]:
consulta = ''' SELECT MAX("Rendimiento Sobre Los Activos (ROA)") AS "ROA Maximo", Año
               FROM "RATIOS DE RENTABILIDAD"
               '''

#Ejecutamos la consulta
cursor.execute(consulta)
#Traemos el resultado de la consulta
Utilidad = cursor.fetchall()
for fila in Utilidad:
  print(fila)

(0.2207, 2022)


Cual es la variacion porcentual entre los activos no corrientes? (Año 2020-2023)

In [ ]:
consulta2020 = ''' SELECT [Activos no corrientes]
               FROM [BALANCE GENERAL]
               WHERE Año = '2020'
               '''

#Ejecutamos la consulta
cursor.execute(consulta2020)
#Traemos el resultado de la consulta
Nocorrientes2020 = cursor.fetchall()
for fila in Nocorrientes2020:
  print(fila)

(3625000,)


In [ ]:
consulta2023 = ''' SELECT [Activos no corrientes]
               FROM [BALANCE GENERAL]
               WHERE Año = '2023'
               '''

#Ejecutamos la consulta
cursor.execute(consulta2023)
#Traemos el resultado de la consulta
Nocorrientes2023 = cursor.fetchall()
for fila in Nocorrientes2023:
  print(fila)

(18109000,)


[(18109000,)]

In [ ]:
# Convert the lists of tuples to numeric values
Nocorrientes2020 = Nocorrientes2020[0][0] # Access the first element of the first tuple
Nocorrientes2023 = Nocorrientes2023[0][0] # Access the first element of the first tuple

consultavariacion = (Nocorrientes2023 - Nocorrientes2020)/Nocorrientes2020 * 100
print(consultavariacion)

399.5586206896552


La variacion porcentual seria de 399.55%

Cual es el patrimonio para el 2023?

In [ ]:
consulta = ''' SELECT [Total de patrimonio]
               FROM [BALANCE GENERAL]
               WHERE Año = '2023'
               '''

#Ejecutamos la consulta
cursor.execute(consulta)
#Traemos el resultado de la consulta
patrimonio = cursor.fetchall()
for fila in patrimonio:
  print(fila)

(22101000,)


Que año tuvo el mayor EBITDA

In [ ]:
consulta = ''' SELECT MAX([EBITDA]) AS "EBITDA Maximo", Año
               FROM [ESTADO DE RESULTADOS]
               '''

#Ejecutamos la consulta
cursor.execute(consulta)
#Traemos el resultado de la consulta
ebitda = cursor.fetchall()
for fila in ebitda:
  print(fila)

(11215000, 2022)


# Apple

Creamos la base de datos

In [ ]:
con = sqlite3.connect('apple.db')
cursor = con.cursor()

Pasamos la informacion del diccionario a la base de datos

In [ ]:
for hoja, apple in apple.items():
  apple.to_sql(hoja, con, if_exists='replace', index=False)

In [ ]:
#Confirmamos los datos
con.commit()

Visualizamos las tablas en formato consulta

In [ ]:
consulta = ''' SELECT name
               FROM sqlite_master
               WHERE type = 'table' '''
#Ejecutamos la consulta
cursor.execute(consulta)
#Printeamos el resultado de la consulta
print(cursor.fetchall())

[('BALANCE  GENERAL',), ('RATIOS DE RENTABILIDAD',), ('ESTADO DE RESULTADOS',), ('ESTADO DE FLUJO DE EFECTIVO',)]


Miramos la informacion de la tabla RATIOS DE RENTABILIDAD

In [ ]:
consulta = ''' SELECT *
              FROM "RATIOS DE RENTABILIDAD" '''
#Ejecutamos la consulta
cursor.execute(consulta)
#Traemos el resultado de la consulta
Ratios = cursor.fetchall()
for fila in Ratios:
  print(fila)

(2020, 0.1773, 0.1773, 0.8787, 0.8787, 0.3057, 0.2091, 0.4177, 0.2433, 0.2768, 0.2062)
(2021, 0.2697, 0.2697, 1.5007, 1.5007, 0.8973, 0.2588, 0.4438, 0.5532, 0.3238, 0.5765)
(2022, 0.2829, 0.2829, 1.9696, 1.9696, 0.5997, 0.2531, 0.4552, 0.3023, 0.325, 0.3379)
(2023, 0.2751, 0.2751, 1.5608, 1.5608, 0.5496, 0.2531, 0.4635, 0.2972, 0.3204, 0.3231)


Lo pasamos a Pandas

In [ ]:
columnas = [descripcion[0] for descripcion in cursor.description]
df_carta = pd.DataFrame(Ratios, columns=columnas)
df_carta

Año  Rendimiento Sobre Los Activos (ROA)  \
0  2020                               0.1773   
1  2021                               0.2697   
2  2022                               0.2829   
3  2023                               0.2751   

   Rendimiento Sobre Activos Anualizado  \
0                                0.1773   
1                                0.2697   
2                                0.2829   
3                                0.2751   

   Rendimiento Sobre El Patrimonio (ROE)  \
0                                 0.8787   
1                                 1.5007   
2                                 1.9696   
3                                 1.5608   

   Rendimiento Sobre El Patrimonio Anualizado  \
0                                      0.8787   
1                                      1.5007   
2                                      1.9696   
3                                      1.5608   

   Rendimiento Sobre El Capital Empleado  Margen Neto  \
0                                 0.3057       0.2091   
1                                 0.8973       0.2588   
2                                 0.5997       0.2531   
3                                 0.5496       0.2531   

   Margen De Ganancia Bruta  Margen Operacional  Margen Ebitda  ROA Operativo  
0                    0.4177              0.2433         0.2768         0.2062  
1                    0.4438              0.5532         0.3238         0.5765  
2                    0.4552              0.3023         0.3250         0.3379  
3                    0.4635              0.2972         0.3204         0.3231

## Consultas

Cual es la utilidad neta total de Apple del 2020 al 2023?

In [ ]:
consulta = ''' SELECT SUM("Utilidad Neta") AS "Utilidad Neta Total"
               FROM "ESTADO DE FLUJO DE EFECTIVO"
               '''

#Ejecutamos la consulta
cursor.execute(consulta)
#Traemos el resultado de la consulta
Utilidad = cursor.fetchall()
for fila in Utilidad:
  print(fila)

(348889000,)


Que año tiene mas rendimiento sobre los activos (ROA)

In [ ]:
consulta = ''' SELECT MAX("Rendimiento Sobre Los Activos (ROA)") AS "ROA Maximo", Año
               FROM "RATIOS DE RENTABILIDAD"
               '''

#Ejecutamos la consulta
cursor.execute(consulta)
#Traemos el resultado de la consulta
Utilidad = cursor.fetchall()
for fila in Utilidad:
  print(fila)

(0.2829, 2022)


Cual es la variacion porcentual entre los activos no corrientes? (Año 2020-2023)

In [ ]:
consulta2020 = ''' SELECT [Activos no corrientes]
               FROM [BALANCE  GENERAL]
               WHERE Año = '2020'
               '''

#Ejecutamos la consulta
cursor.execute(consulta2020)
#Traemos el resultado de la consulta
Nocorrientes2020 = cursor.fetchall()
for fila in Nocorrientes2020:
  print(fila)

(180175000,)


In [ ]:
consulta2023 = ''' SELECT [Activos no corrientes]
               FROM [BALANCE  GENERAL]
               WHERE Año = '2023'
               '''

#Ejecutamos la consulta
cursor.execute(consulta2023)
#Traemos el resultado de la consulta
Nocorrientes2023 = cursor.fetchall()
for fila in Nocorrientes2023:
  print(fila)

(209017000,)


In [ ]:
# Convert the lists of tuples to numeric values
Nocorrientes2020 = Nocorrientes2020[0][0] # Access the first element of the first tuple
Nocorrientes2023 = Nocorrientes2023[0][0] # Access the first element of the first tuple

consultavariacion = (Nocorrientes2023 - Nocorrientes2020)/Nocorrientes2020 * 100
print(consultavariacion)

16.007770223393923


La variacion porcentual seria del 16%

Cual es el patrimonio para el 2023?

In [ ]:
consulta = ''' SELECT [Total de patrimonio]
               FROM [BALANCE  GENERAL]
               WHERE Año = '2023'
               '''

#Ejecutamos la consulta
cursor.execute(consulta)
#Traemos el resultado de la consulta
patrimonio = cursor.fetchall()
for fila in patrimonio:
  print(fila)

(62146000,)


Que año tuvo el mayor EBITDA

In [ ]:
consulta = ''' SELECT MAX([EBITDA]) AS "EBITDA Maximo", Nombre
               FROM [ESTADO DE RESULTADOS]
               '''

#Ejecutamos la consulta
cursor.execute(consulta)
#Traemos el resultado de la consulta
ebitda = cursor.fetchall()
for fila in ebitda:
  print(fila)

(128137000, 2022)


# Comparacion

### Utilidad Neta Total



*   NVIDIA: 21'248,000
*   Apple: 348'889,000

Ambos estados financieros estan en la misma escala (Miles). Podemos ver que Apple del 2020 al 2023 obtuvo una mayor utilidad neta total.


### Rendimiento Sobre los Activos (ROA)

*   NVIDIA: En el 2022 tuvo 22% de ROA
*   Apple: En el 2022 tuvo un 28% de ROA

Ambas compañías tuvieron su ROA máximo en el 2022 pero Apple también superó a NVIDIA.

### Variacion Porcentual entre los Activos No Corrientes (2020-2023)

*   NVIDIA: 399.55%
*   Apple: 16%

La variación porcentual de los activos no corrientes entre el 2020 y 2023 fue positiva para ambas compañías. Sin embargo, NVIDIA tuvo un crecimiento anormal de sus activos no corrientes. Superando de lejos a Apple.

### Patrimonio para el 2023

*   NVIDIA: 22'101,000
*   Apple: 62'146,000

Para el 2023, Apple superó a NVIDIA en patrimonio total

### EBITDA

*   NVIDIA (2022): 11'215,000
*   Apple (2022): 128'137,000

En el 2022, ambas compañías tuvieron su EBITDA máximo, pero Apple supera de lejos a NVIDIA


### Conclusion

De las variables comparadas, podríamos decir que Apple es una compañía mucho mas grande. Sin embargo, al ver el crecimiento porcentual de los activos no corrientes de NVIDIA podemos ver que es una compañía que ha crecido bastante.